include/image.h

```diff
@@ -125,22 +157,42 @@ class Image
             + ")";
     }
 
+    const bool operator == (const Image &other) const {
+        if(images_compatible<t, C, t, C>(*this, other)) {
+            const size_t n_rows = get_n_rows();
+            const size_t n_cols = get_n_cols();
+            for(size_t row = 0; row < n_rows; ++row) {
+                for(size_t col = 0; col < n_cols; ++col) {
+                    if(this->get_value_unchecked(row, col) != other.get_value_unchecked(row, col)) {
+                        return false;
+                    }
+                }
+            }
+            return true;
+        }
+        return false;
+    }
+
+    const bool operator != (const Image &other) const {
+        return !(*this == other);
+    }
+
+    // TODO: Figure out if there's any point to this in CRTP (or otherwise)
```

Comment:
I'm not sure what the questions was, but if you forgot about this TODO and want to discuss it here I'd be happy to.

Reply:
I think my intention was to have a deleted constructor to force subclasses to implement it, and I wasn't sure if that actually worked in CRTP (whereas it should in non-templated inheritance, no?).